jobs/integration/validation.py

```diff
@@ -1947,11 +1947,7 @@ async def get_offer():
             coredns = k8s_model.applications["coredns"]
 
         log("Waiting for CoreDNS charm to be ready")
-        while (
-            status := coredns.units[0].workload_status if coredns.units else None
-        ) != "active":
-            assert status not in ("blocked", "error")
-            await asyncio.sleep(5)
+        await k8s_model.wait_for_idle(raise_on_error=False, status="active")
```

Comment:
I think we should add a comment here so that we can come back later when this issue is addressed.

Reply:
i'm guessing the offensive (workaroundy) part here is the `raise_on_error=False`

this prevents the method from raising an exception when ANY error occurs.  Now, if the charm isn't active after the standard timeout (10m) then if it's still in error, a timeout error will occur

Reply:
Alright, that makes sense! :)